lucupy/minimodel/group.py

```diff
@@ -43,39 +43,37 @@ class Group(ABC):
     number_to_observe: int
     delay_min: timedelta
     delay_max: timedelta
-
-    # Cannot be frozen with gelidum.
     children: Union[List['Group'], Observation]
 
+    # Calculated afterward.
+    unique_id: UniqueGroupID = field(init=False)
+
     def __post_init__(self):
         if self.number_to_observe <= 0:
             msg = f'Group {self.group_name} specifies non-positive {self.number_to_observe} children to be observed.'
             raise ValueError(msg)
 
-    def unique_id(self) -> UniqueGroupID:
-        """Get a unique name for the group.
-        If this is an observation group, it is just the group_id.
-        If this is a scheduling group, since in OCS this is just an integer, we combine with the program_id to
-            make it unique.
-
-        Returns:
-            UniqueGroupID: A unique identifier for the group.
-        """
-        if self.id.startswith(self.program_id):
-            return self.id
+        if self.id.id.startswith(self.program_id.id):
```

Comment:
self.id.id looks weird but I understand why is there

Reply:
It felt even stranger to type, but that logic should be confined to lucupy.